### PR TITLE
Filter endpoints by type

### PR DESCRIPTION
**Pull request by richard-cox**
- Add endpoint type filter to endpoint page list 
- fixes #4563
- fixes #4565

### CODE REVIEW

File: src/frontend/packages/core/src/shared/components/list/list-types/endpoint/base-endpoints-data-source.ts [hunk 1]
```diff
@@ -73,21 +76,28 @@ export class BaseEndpointsDataSource extends ListDataSource<EndpointModel> {
...
+      transformEntities.push((entities: EndpointModel[]) => {
+        return dsEndpointType || onlyConnected ? entities.filter(endpoint => {
+          return (!onlyConnected || endpoint.connectionStatus === 'connected') &&
+            (!dsEndpointType || endpoint.cnsi_type === dsEndpointType);
+        }) : entities;
+      });
+    }
+    if (filterByType) {
+      transformEntities.push((entities: EndpointModel[], paginationState: PaginationEntityState) =>
+        BaseEndpointsDataSource.endpointTypeFilter(entities, paginationState)
```
Comment (richard-cox) [inline src/frontend/packages/core/src/shared/components/list/list-types/endpoint/base-endpoints-data-source.ts:96]: Shouldn't be static

File: src/frontend/packages/core/src/shared/components/list/list-types/endpoint/endpoint-list.helpers.ts [hunk 2]
```diff
@@ -48,6 +49,8 @@ export class EndpointListHelper {
     private snackBarService: SnackBarService,
   ) { }
 
+  public static typeFilterKey = 'endpointType';
```
Comment (richard-cox) [inline src/frontend/packages/core/src/shared/components/list/list-types/endpoint/endpoint-list.helpers.ts]: Move to the data source

File: src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-reset-pagination.ts [hunk 3]
```diff
@@ -102,3 +103,19 @@ function paginationResetPaginationState(oldEntityState: PaginationEntityState) {
   }
   return entityState;
 }
+
+export function resetEndpointEntities(state: PaginationState, action: EndpointActionComplete) {
```
Comment (richard-cox) [inline src/frontend/packages/store/src/reducers/pagination-reducer/pagination-reducer-reset-pagination.ts:107]: Bug, unregister cf, go to cf endpoints page, no sort order

**Review by nwmac — APPROVED** [review]
LGTM - just a couple of things to check

**Review by richard-cox — COMMENTED** [review]
See comments

File: src/frontend/packages/core/src/shared/components/list/list-types/endpoint/base-endpoints-data-source.ts (hunk 1 above)
Comment (richard-cox) [reply src/frontend/packages/core/src/shared/components/list/list-types/endpoint/base-endpoints-data-source.ts:96]: Calling this before a super in the ctor, cannot do that if method is not static

**Pull request merged**